services/skus/receipt.go

```diff
@@ -145,54 +137,45 @@ func (v *receiptVerifier) validateApple(ctx context.Context, req model.ReceiptRe
 	return resp.Receipt.InApp[0].OriginalTransactionID, nil
 }
 
+const (
+	errPurchasePending model.Error = "purchase pending"
+	errPurchaseExpired model.Error = "purchase expired"
+)
+
 // validateGoogle validates Google Store receipt.
 func (v *receiptVerifier) validateGoogle(ctx context.Context, req model.ReceiptRequest) (string, error) {
-	l := logging.Logger(ctx, "skus").With().Str("func", "validateReceiptGoogle").Logger()
-
-	l.Debug().Str("receipt", fmt.Sprintf("%+v", req)).Msg("about to verify subscription")
-
-	resp, err := v.playStoreCl.VerifySubscription(ctx, req.Package, req.SubscriptionID, req.Blob)
+	resp, err := v.androidSubPurchaser.GetSubscriptionPurchase(ctx, req.Package, req.SubscriptionID, req.Blob)
 	if err != nil {
-		l.Error().Err(err).Msg("failed to verify subscription")
-		return "", errPurchaseFailed
+		return "", fmt.Errorf("error retrieving subscription purchase: %w", err)
 	}
 
-	// Check order expiration.
-	if time.Unix(0, resp.ExpiryTimeMillis*int64(time.Millisecond)).Before(time.Now()) {
+	if isExpired(resp.ExpiryTimeMillis, time.Now()) {
 		return "", errPurchaseExpired
 	}
 
-	l.Debug().Msgf("resp: %+v", resp)
-
-	if resp.PaymentState == nil {
-		l.Error().Err(err).Msg("failed to verify subscription: no payment state")
-		return "", errPurchaseFailed
+	if isSubPurchasePending(resp) {
+		return "", errPurchasePending
 	}
 
-	// Check that the order was paid.
-	switch *resp.PaymentState {
-	case androidPaymentStatePaid, androidPaymentStateTrial:
-		return req.Blob, nil
-
-	case androidPaymentStatePending:
-		// Check for cancel reason.
-		switch resp.CancelReason {
-		case androidCancelReasonUser:
-			return "", errPurchaseUserCanceled
-		case androidCancelReasonSystem:
-			return "", errPurchaseSystemCanceled
-		case androidCancelReasonReplaced:
-			return "", errPurchaseReplacedCanceled
-		case androidCancelReasonDeveloper:
-			return "", errPurchaseDeveloperCanceled
-		}
-		return "", errPurchasePending
+	return req.Blob, nil
+}
 
-	case androidPaymentStatePendingDeferred:
-		return "", errPurchaseDeferred
-	default:
-		return "", errPurchaseStatusUnknown
+func isExpired(expTimeMills int64, now time.Time) bool {
+	return now.UnixMilli() > expTimeMills
+}
```

Comment:
suggestion: The name seems a bit too generic. We could make it more specific in the context of time it is dealing with, or the task it performs:

1. A more time-specific option:
```golang
func hasExpiredMilli(now, target int64) bool {
	return now > target
}
```

2. A more task-specific name:
```golang
func hasPlaySubExpired(now time.Time, targetMilli int64) bool {
	return now.Milli() > targetMilli
}
```

Reply:
update so its more specific